src/Workleap.OpenApi.MSBuild/SwaggerManager.cs

```diff
@@ -0,0 +1,70 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SwaggerManager : ISwaggerManager
+{
+    private const string SwaggerVersion = "6.5.0";
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiWebApiAssemblyPath;
+    private readonly string _swaggerDirectory;
+
+    public SwaggerManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, string openApiWebApiAssemblyPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiWebApiAssemblyPath = openApiWebApiAssemblyPath;
+        this._swaggerDirectory = Path.Combine(openApiToolsDirectoryPath, $"swagger/{SwaggerVersion}");
+    }
+
+    public async Task RunSwaggerAsync(string[] openApiSwaggerDocumentNames, CancellationToken cancellationToken)
+    {
+        var swaggerExePath = Path.Combine(this._swaggerDirectory, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "swagger.exe" : "swagger");
+
+        foreach (var documentName in openApiSwaggerDocumentNames)
+        {
+            var outputOpenApiSpecName = $"openapi-{documentName.ToLowerInvariant()}.yaml";
+
+            // IDP-686, this is not the right path, should be in another folder named reports
```

Comment:
I don't think we have to save the generated spec file in `Reports` based on what we have on Slack Canvas